### PR TITLE
add intensity, ring index support in ply loader

### PR DESCRIPTION
**Pull request by johnwlambert**
Provides a function to load not only x,y,z coordinates for each point in an Argoverse LiDAR sweep, but also to load reflectance/intensity and the ring index.

Thanks to @irenecortes for this update from her repo (Ref: https://github.com/irenecortes/argoverse_lidar), and to @apak-00 for the function name.

### CODE REVIEW

**Comment by benjaminrwilson** [comment]
Not sure what the status of this is -- but this looks promising as well: http://www.open3d.org/docs/latest/python_api/open3d.t.io.read_point_cloud.html#open3d.t.io.read_point_cloud (if this is ever considered being added as a dependency).

File: argoverse/utils/ply_loader.py [hunk 1]
```diff
@@ -26,3 +26,23 @@ def load_ply(ply_fpath: _PathLike) -> np.ndarray:
     z = np.array(data.points.z)[:, np.newaxis]
 
     return np.concatenate((x, y, z), axis=1)
+
+
+def load_ply_xyzil(ply_fpath: _PathLike) -> np.ndarray:
```
Comment (johnwlambert) [inline argoverse/utils/ply_loader.py:31]: maybe pass in string "xyz" or "xyzi" to determine what channels you want

File: argoverse/utils/ply_loader.py [hunk 2]
```diff
@@ -26,3 +26,23 @@ def load_ply(ply_fpath: _PathLike) -> np.ndarray:
...
+
+    Returns:
+        arr: Array of shape (N, 5), i.e. x,y,z,intensity,laser_number
+    """
+
+    data = pyntcloud.PyntCloud.from_file(os.fspath(ply_fpath))
+    x = np.array(data.points.x)[:, np.newaxis]
+    y = np.array(data.points.y)[:, np.newaxis]
+    z = np.array(data.points.z)[:, np.newaxis]
+    i = np.array(data.points.intensity)[:, np.newaxis]
```
Comment (johnwlambert) [inline argoverse/utils/ply_loader.py:45]: make list comprehension from dictionary

**Comment by venice-erin** [comment]
hi, any reason this haven't been merged yet?

File: argoverse/utils/ply_loader.py (hunk 2 above)
Comment (johnwlambert) [reply argoverse/utils/ply_loader.py]: added

File: argoverse/utils/ply_loader.py (hunk 1 above)
Comment (johnwlambert) [reply argoverse/utils/ply_loader.py]: added

File: argoverse/utils/ply_loader.py [hunk 3]
```diff
@@ -26,3 +26,37 @@ def load_ply(ply_fpath: _PathLike) -> np.ndarray:
     z = np.array(data.points.z)[:, np.newaxis]
 
     return np.concatenate((x, y, z), axis=1)
+
+
+def load_ply_by_attrib(ply_fpath: _PathLike, attrib_spec: str) -> Optional[np.ndarray]:
```
Comment (benjaminrwilson) [inline argoverse/utils/ply_loader.py:31]: Should we default `attrib_spec` to `attrib_spec = "xyzil"`?

Comment (johnwlambert) [reply argoverse/utils/ply_loader.py:31]: i like that

Comment (johnwlambert) [reply argoverse/utils/ply_loader.py]: updated

File: argoverse/utils/ply_loader.py [hunk 4]
```diff
@@ -26,3 +26,37 @@ def load_ply(ply_fpath: _PathLike) -> np.ndarray:
...
+            x -> point x-coord
+            y -> point y-coord
+            z -> point z-coord
+            i -> point intensity/reflectance
+            l -> laser number of laser from which point was returned
+
+    Returns:
+        arr: Array of shape (N, C). If attrib_str is invalid, `None` will be returned
+    """
+    possible_attributes = ["x", "y", "z", "i", "l"]
```
Comment (benjaminrwilson) [inline argoverse/utils/ply_loader.py:46]: I wonder if we should do this with an `OrderedDict`? Probably doesn't matter too much, but we could do something like:

```
possible_attributes = OrderedDict({"x": None, "y": None, "z": None, "i": None, "l": None})
filtered_attributes = list(filter(lambda x: x in possible_attributes, list(attrib_spec)))
if not filtered_attributes:
    return None

data = pyntcloud.PyntCloud.from_file(os.fspath(ply_fpath))
return np.stack([getattr(data.points, name) for name in filtered_attributes], axis=1)
```

Just a thought, though.

Comment (benjaminrwilson) [reply argoverse/utils/ply_loader.py:46]: Actually, I think `OrderedDict` and just be a `Dict`.

**Review by benjaminrwilson — APPROVED** [review]
👍

**Comment by johnwlambert** [comment]
> hi, any reason this haven't been merged yet?

Hi @venice-erin, we were still discussing the API a bit now but we'll merge it now after we've made some updates.

**Comment by venice-erin** [comment]
Thanks :)

**Pull request merged**